core/app/services/spree/products/translations/generate_automated_translations.rb

```diff
@@ -0,0 +1,69 @@
+module Spree
+  module Products
+    module Translations
+      class GenerateAutomatedTranslations
+        prepend Spree::ServiceModule::Base
+
+        def initialize(provider: self.class.injected_automated_translations_provider)
```

Comment:
Calling `self.class.injected_automated_translations` feels a bit clunky to me. I wonder if having a private method like this would work just as well and look cleaner:
```ruby
def automated_translations_provider
  @automated_translations_provider ||= Spree::Dependencies.products_automated_translations_provider&.constantize
end
```
Then we also would not need to set the provider in the initializer. Something to consider.